src/Microsoft.DotNet.ServiceFabric.ServiceHost/DelegatedService.cs

```diff
@@ -0,0 +1,106 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.ServiceFabric.ServiceHost
+{
+    public static class DelegatedService
+    {
+        public static async Task RunServiceLoops<T>(IServiceProvider services, CancellationToken cancellationToken, params Func<CancellationToken, Task>[] loops)
+        {
+            using var linkedToken = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            await Lifecycle.OnStartingAsync(services);
+            var logger = services.GetRequiredService<ILogger<T>>();
+            List<Task> pendingTasks = new List<Task>();
+            try
+            {
+                await using var _ =
+                    cancellationToken.Register(() => logger.LogInformation("Service abort cancellation requested"));
+                logger.LogInformation("Entering service 'RunAsync'");
```

Comment:
Just to be sure, the logger will also capture the name of the service these tasks belong to, correct?

Reply:
Yup, you get a ton of "free" context data whenver you LogSomething.  Basically everything in app insights that isn't the "message" gets tacked on by one plugin or another.